pkg/compact/compact_test.go

```diff
@@ -174,3 +177,241 @@ func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
 	}
 
 }
+
+func TestPlanSimulate(t *testing.T) {
+	planner := NewTSDBBasedPlanner(log.NewNopLogger(), []int64{
+		int64(1 * time.Hour / time.Millisecond),
+		int64(2 * time.Hour / time.Millisecond),
+		int64(8 * time.Hour / time.Millisecond),
+		int64(2 * 24 * time.Hour / time.Millisecond),
+	})
+	reg := prometheus.NewRegistry()
+	ps := NewCompactionSimulator(reg, planner)
+
+	metas := []*metadata.Meta{
```

Comment:
Can you add more test cases and make it a table like you do in the downsample tests?

Reply:
Yes, I'll organise and add more test cases for both the unit tests once everything else is done. Since I know how to do table testing now, it shouldn't take long.

Reply:
I've added some more cases and organised TestPlanSimulate using table tests. I'll continue adding more cases. PTAL and let me know if that is okay.
Thanks!